### PR TITLE
Extract edit history to separate pallets

### PR DESCRIPTION
**Pull request by F3Joule**

### CODE REVIEW

File: pallets/utils/src/lib.rs [hunk 1]
```diff
@@ -39,17 +39,24 @@ pub enum User<AccountId> {
     Space(SpaceId),
 }
 
+#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug)]
+pub enum ContentType {
```
Comment (subsocialdev) [inline pallets/utils/src/lib.rs:43]:
```suggestion
pub enum Content {
```

File: pallets/utils/src/lib.rs [hunk 2]
```diff
@@ -39,17 +39,24 @@ pub enum User<AccountId> {
...
+    None,
+    Ipfs(Vec<u8>),
```
Comment (subsocialdev) [inline pallets/utils/src/lib.rs:45]:
```suggestion
    IPFS(Vec<u8>),
```

File: pallets/utils/types.json [hunk 3]
```diff
@@ -12,5 +12,13 @@
       "Account": "AccountId",
       "Space": "SpaceId"
     }
+  },
+
+  "ContentType": {
+    "_enum": {
+      "None": "Null",
+      "Ipfs": "Text",
```
Comment (subsocialdev) [inline pallets/utils/types.json:20]:
```suggestion
  "Content": {
    "_enum": {
      "None": "Null",
      "IPFS": "Text",
```

File: types.json [hunk 4]
```diff
@@ -15,6 +14,14 @@
     }
   },
 
+  "ContentType": {
+    "_enum": {
+      "None": "Null",
+      "Ipfs": "Text",
```
Comment (subsocialdev) [inline types.json:20]:
```suggestion
  "Content": {
    "_enum": {
      "None": "Null",
      "IPFS": "Text",
```

**Comment by F3Joule** [comment]
At this point, all commits are based on current `master` branch.

**Comment by F3Joule** [comment]
#37 should be merged first

**Comment by F3Joule** [comment]
Broken.

**Pull request closed without merging**